packages/module-loader-commonjs/src/index.ts

```diff
@@ -9,7 +11,15 @@ export function createCommonJsModuleLoader(
   externals: Externals = {}
 ): ModuleLoader {
   return async (url: string): Promise<unknown> => {
-    const response = await fetch(url);
+    const requestInit: RequestInit = {};
+
+    if (process.env.HTTP_PROXY) {
+      requestInit.agent = (new ProxyAgent(
+        process.env.HTTP_PROXY
+      ) as unknown) as Agent;
```

Comment:
This type cast caught my attention. Apparently only the latest version of `agent-base` matches the `Agent` interface of Node. This version is [not yet used in `proxy-agent`](https://github.com/TooTallNate/node-proxy-agent/issues/51). But it is used in `http-proxy-agent` as well as `https-proxy-agent`. Which made me think why you called the env `HTTP_PROXY`. Do you only need http, or should this also cover https (probably)? ... Does the feature hub even need to handle all of this? We already have a factory function for the CommonJS module loader, so why not pass the agent as an optional second param here? Then you can use whatever proxy lib you need in the integrator (still based on an env variable if you like). Maybe we should even allow the whole `requestInit` object to be passed instead of only the `agent`? What do you think, @clebert?

Reply:
Yeah, that's a good idea. This allows for greater flexibility on the integrator side. So I would be in favor of passing the whole requestInit object in.

Reply:
Hi @unstubbable ,
Hi @clebert ,

Thanks for your feedback!

So, creating a feature hub on our side would look like:

```javascript
const requestInit: RequestInit = {};

if (process.env.HTTP_PROXY) {
  requestInit.agent = (new ProxyAgent(
    process.env.HTTP_PROXY
  ) as unknown) as Agent;
}

createFeatureHub('integrator-id', {
  // ...
  moduleLoader: createCommonJsModuleLoader(requestInit),
  // ...
});
```

Right?

How would you like to pass `requestInit` to `createCommonJsModuleLoader`? I'd prefer an options object since it is extendable in the future, like:
```javascript
export interface Options {
  externals: Externals,
  requestInit: RequestInit
}

export function createCommonJsModuleLoader(options: Options): ModuleLoader
```

But this is a breaking change. We could also add a second parameter?

It's up to you :)

Kind regards
Niels

Reply:
As I said, use a second param so we are not breaking.

Reply:
Hi @unstubbable , Hi @clebert ,
I added the second optional parameter. I squashed the commits and changed the commit message since the scope changed.